lib/source-code/token-store/utils.js

```diff
@@ -77,13 +82,18 @@ exports.getLastIndex = function getLastIndex(tokens, indexMap, endLoc) {
     }
     if ((endLoc - 1) in indexMap) {
         const index = indexMap[endLoc - 1];
-        const token = (index >= 0 && index < tokens.length) ? tokens[index] : null;
+        const token = tokens[index];
+
+        // If the mapped index is out of bounds, the returned cursor index will point before the start of the tokens array.
+        if (!token) {
```

Comment:
Same as https://github.com/eslint/eslint/pull/16889/files#r1131283705. `index` can exceed the array length, but it should never be negative.